### DISCUSSION TITLE
tests helpers class

### DISCUSSION THREAD

**Discussion by samcoppock** [wintercms/winter#1180] [General]
In a previous version of winter this file existed.

"tests/bootstrap.php"

In our project we have to use it or else the test dont work and we append it using 

"-dauto_prepend_file=" when running tests manually on the CLI using the phpunit command.

In the most recent version of Winter "tests/bootstrap.php" no longer exists and appears to have been replaced with "modules/system/tests/bootstrap/app.php"

I assumed that the new "php artisan winter:test" command would automatically bootstrap this file.

But it does not - I know this because I placed die statements in the file and none of them activated.
And of course even with the die statements removed the tests failed because the bootstrap file was not loaded.

Also when running the tests using the php unit command I have to use "-dauto_prepend_file=" to add ether the new "modules/system/tests/bootstrap/app.php" file or the old "tests/bootstrap.php" file ( which we still have in our codebase ) in order to make the tests work. ( that is to say both versions of the file still work - although the old one has had some of the new ones code copy/pasted into it)

Which leads to my question

How do I include "modules/system/tests/bootstrap/app.php" when running tests using "php artisan winter:test" to run tests?

**Comment by mjauvin** [comment]
Which tests are you running ? The bootstrap file is included when extending `System\Tests\Bootstrap\TestCase`.

One example is `Backend\Tests\Classes\AuthManagerTest.php`

**Reply by samcoppock** [reply]
we have some unit tests which extend the old version of TestCase and some feature tests which currently extend the old version of PluginTestCase ( thoes ones actually extend a class we made that has some of our own code which in turn extends Plugintestcase )

of course the PluginTestCase.php ones by extension extend TestCase.php

I tried modifying our custom class to extend the new version of PlugintestCase but i have not yet modified the unit tests to extend the new version of TestCase.php

This works.
Until I delete the old bootstrap.php file

at which point i get something like this

Unable to find the bootstrap file "/var/www/html/tests/bootstrap.php"

the wording is different depending on if i use phpunit or the new artisan command to run tests but it refers to the same file

the error appears before any tests run.

I am guessing this means i need to also update the Testcase.php referances in the unit tests to point to the new version of that file as well before any tests will work.

**Reply by LukeTowers** [reply]
@samcoppock did you read my answer in #1177?

**Reply by samcoppock** [reply]
yes i read 1177
it did not seam to be relevant
